src/FakeItEasy/Creation/DummyValueResolver.cs

```diff
@@ -190,6 +191,58 @@ private static Func<T> CreateFunc<T>(T value)
             }
         }
 
+        private class ResolveByCreatingTupleStrategy : ResolveStrategy
+        {
+            private readonly ConcurrentDictionary<Type, Func<object[], object>> tupleConstructorsCache =
+                new ConcurrentDictionary<Type, Func<object[], object>>();
+
+            public override CreationResult TryCreateDummyValue(Type typeOfDummy, IDummyValueResolver resolver, LoopDetectingResolutionContext resolutionContext)
+            {
+                if (IsTuple(typeOfDummy, "System.Tuple`") || IsTuple(typeOfDummy, "System.ValueTuple`"))

```

Comment:
`IsTuple` feels too specific for a method that still needs to be told "System.Tuple" or "System.ValueTuple". And running this way, we're checking the generic type definition twice. I'd keep the method name and just pass in `typeOfDummy` and do the check against the two strings inside.